src/Terminal/ui/TerminalInput.tsx

```diff
@@ -301,10 +296,10 @@ export function TerminalInput(): React.ReactElement {
       if (Settings.EnableBashHotkeys) {
         event.preventDefault();
       }
-      if (searchResultsIndex !== -1) {
-        const index = Math.max(searchResultsIndex - 1, 0);
-        setSearchResultsIndex(index);
-        setCurrentSearchPossibility(searchResults[index]);
+      if (searchResults.length > 1) {
+        const results = [...searchResults];
+        results.unshift(results.pop() ?? "");
+        setSearchResults(results);
```

Comment:
do you need a shallow copy of the searchResults? 

I might suggest using the setState's callback fn. 
```suggestion
        setSearchResults((results) => {
            results.unshift(results.pop() ?? "");
            return results;
        });
```





technically you could also do 
```suggestion
        searchResults.unshift(searchResults.pop() ?? "");
        setSearchResults(searchResults);
```
though mutating state is usually considered tabboo